parser/ast.go

```diff
@@ -182,3 +191,12 @@ func toEnumElements[T ast.EnumElement](nodes nodeWithEmptyDecls[T]) []ast.EnumEl
 	}
 	return elements
 }
+
+func toMessageElements[T ast.MessageElement](nodes nodeWithEmptyDecls[T]) []ast.MessageElement {
```

Comment:
Sadly, we can't make this more generic and collapse into a single function, since there's no way to express assignability/convertibility constraints in Go generic type parameters :/

But we could at least hoist this body into another generic function so each of these functions becomes a one-liner.